File: Lab/Cryptography and Network Security Lab/Encryption-Decryption.py
pltx = "iambadboycode"
k = 12
enc = ""

# ---------- Encryption ----------
for i in pltx:
    singleChar = ord(i) - ord('a')
    singleChar += k
    singleChar = singleChar % 26
    singleChar += ord('a')
    enc += chr(singleChar)

print("Encrypted text:", enc)



sampleArray = ['me','very','good','table','box','not','possible','well','book','room']

found_k = None

for kx in range(26):  
    decoded = ""

    for i in enc:
        singledecode = ord(i) - ord('a')
        singledecode -= kx         
        singledecode = singledecode % 26
        singledecode += ord('a')
        decoded += chr(singledecode)

    
    for word in sampleArray:
        if word in decoded:
            found_k = kx
            print("Decoded text:", decoded)
            break

    if found_k is not None:
        break

print(f"Found key = {found_k}")

File: Lab/Cryptography and Network Security Lab/Encryption-Decryption.cpp
#include <bits/stdc++.h>
using namespace std;


string encrypt(string text, int key) {
    string result = "";
    for (char c : text)
        result += char((c - 'a' + key) % 26 + 'a');
    return result;
}


string decrypt(string text, int key) {
    string result = "";
    for (char c : text)
        result += char((c - 'a' - key + 26) % 26 + 'a');
    return result;
}

int main() {
    string p, d;
    int k;

    cout << "Enter message: ";
    cin >> p;

    cout << "Enter key: ";

    cin >> k;

    string e = encrypt(p, k);
    cout << "Encrypted text: " << e << endl;

    string w[10] = {"me","very","good","code","box","not","possible","well","book","room"};

    for (int x = 0; x < 26; x++) {
        d = decrypt(e, x);
        for (int i = 0; i < 10; i++) {
            if (d.find(w[i]) != -1) {
                cout << "Decoded text: " << d << endl;
                cout << "Decryption key: " << x << endl;
                return 0;
            }

        }
    }

    return 0;
}
